scripts/policy/frameworks/cluster/controller/main.zeek

```diff
@@ -333,51 +376,37 @@ event ClusterController::API::get_instances_request(reqid: string)
 
 	local insts: vector of ClusterController::Types::Instance;
 
-	for ( i in ClusterController::instances )
-		insts += ClusterController::instances[i];
+	for ( i in g_instances )
+		insts += g_instances[i];
 
 	ClusterController::Log::info(fmt("tx ClusterController::API::get_instances_response %s", reqid));
 	event ClusterController::API::get_instances_response(reqid, insts);
 	}
 
 event zeek_init()
 	{
-	# Controller always listens -- it needs to be able to respond
-	# to the Zeek client. This port is also used by the agents
-	# if they connect to the client.
+	# The controller always listens -- it needs to be able to respond to the
+	# Zeek client. This port is also used by the agents if they connect to
+	# the client. The client doesn't automatically establish or accept
+	# connectivity to agents: agents are defined and communicated with as
+	# defined via configurations defined by the client.
+
 	local cni = ClusterController::network_info();
+
 	Broker::listen(cat(cni$address), cni$bound_port);
 
 	Broker::subscribe(ClusterAgent::topic_prefix);
 	Broker::subscribe(ClusterController::topic);
 
+	# Events sent to the client:
+
 	Broker::auto_publish(ClusterController::topic,
 	    ClusterController::API::get_instances_response);
 	Broker::auto_publish(ClusterController::topic,
 	    ClusterController::API::set_configuration_response);
 
-	if ( |ClusterController::instances| > 0 )
```

Comment:
There's no initial peering anymore at controller startup now. Is that because we won't have any instances at first now? Once we get config persistence, we'll get that back I suppose?

Reply:
Yes exactly — with persistence, there'll be something to do, right now it all waits until you push a config.